src/config.rs

```diff
@@ -133,6 +138,18 @@ impl Config {
             config.excerpt_separator = excerpt_separator.to_owned();
         };
 
+        #[cfg(all(feature="syntax-highlight", not(windows)))]
+        {
+            if let Some(theme) = yaml["syntax-highlight"]["theme"].as_str() {
+                if has_theme(theme) {
+                    config.theme = theme.to_owned();
+                } else {
+                    warn!("Theme named '{}' is not loaded. Using default 'base16-ocean.dark'.",
+                          theme);
+                }
+            };
+        }
+
```

Comment:
Possible fix in commit 'Isolate platform checks for `has_syntax_theme`'.

There's a bit more code to it, but I like the explicitness.  If the platform is supported, return `Ok(bool)`, otherwise return an `Err(reason)`.  An additional benefit is the user is notified why an unsupported feature they're trying to use doesn't work.  Using an `Option` reduces the amount of code in the patch, but seemed like a [possibly] appropriate use of `Option`.  `Option` was definitely less explicit in it's intent.

Overall, the commit is not as succinct as returning a `vec![]` was with `list_syntax_themes`, but I wanted to put this implementation on the table in case you find it preferable.  If so, I'm open to refactoring `list_syntax_themes` to use `Result` for consistency.